jooq/src/main/java/io/micronaut/configuration/jooq/JooqConfigurationFactory.java

```diff
@@ -0,0 +1,53 @@
+package io.micronaut.configuration.jooq;
+
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.annotation.EachBean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Parameter;
+import io.micronaut.inject.qualifiers.Qualifiers;
+import org.jooq.*;
+import org.jooq.conf.Settings;
+import org.jooq.impl.DefaultConfiguration;
+import org.jooq.impl.DefaultDSLContext;
+
+import javax.sql.DataSource;
+
+@Factory
+public class JooqConfigurationFactory {
+
+    @EachBean(DataSource.class)
+    public DefaultConfiguration jooqConfiguration(
+            @Parameter String name,
+            DataSource dataSource,
+            ApplicationContext ctx
+    ) {
+        DefaultConfiguration configuration = new DefaultConfiguration();
+
+        JooqConfigurationProperties properties = ctx.findBean(JooqConfigurationProperties.class, Qualifiers.byName(name))
+                .orElseGet(JooqConfigurationProperties::new);
+        configuration.setSQLDialect(properties.determineSqlDialect(dataSource));
```

Comment:
This might work depending on the data source, but perhaps, at this point, it is not ready yet to produce a JDBC `Connection`. To stay safe, it would be better to allow users to specify the `SQLDialect` explicitly, and execute the above logic only in the absence of an explicit `SQLDialect`.

Explicit `SQLDialect` configurations are necessary anyway, in case the dialect cannot be derived from the JDBC connection (e.g. usage non-standard drivers, third party URL proxies, MySQL or PostgreSQL compatible dialects that jOOQ doesn't officially support (e.g. CockroachDB), etc. etc.)

Reply:
`SQLDialect` can be specified manually via `JooqConfigurationProperties`, I copied this part of logic from Spring Boot integration.

Reply:
I see, I overlooked that the "determine" logic is only executed when the dialect is absent from that properties object.